src/ZoneServer/Skills/Handlers/Swordsman/Peltasta/ShieldLob.cs

```diff
@@ -0,0 +1,213 @@
+			await Task.Delay(TimeSpan.FromMilliseconds(500));
+
+			// Return trip back to player.
+			movement.SetFixedMoveSpeed(200f);
+			cancelToken.Cancel();
+			cancelToken = new CancellationTokenSource();
+			checkShieldCollision(skill, caster, shieldMonster, cancelToken);
+
+			// We loop over this a few times so the shield homes in on your location even if you're moving
+			for (int i = 0; i < 5; i++) 
+			{ 
+				moveTime = movement.MoveTo(caster.Position);
+				Send.ZC_NORMAL.Skill_EffectMovement(caster, skillHandle, caster.Position, 200);
+				await Task.Delay(moveTime.Divide(3));
+			}
+
+			// Should be close enough to your position, return to the player
+			moveTime = movement.MoveTo(caster.Position);
+			Send.ZC_NORMAL.Skill_EffectMovement(caster, skillHandle, caster.Position, 200);
+			await Task.Delay(moveTime);
+
+			// Shield has returned, remove the shield and restore the shield to your hand
+			cancelToken.Cancel();
+			Send.ZC_NORMAL.SkillPad(caster, skill, "Peltasta_ShieldLob2", originPos, caster.Direction, 0, 145.8735f, skillHandle, 30, false);
+			caster.Map.RemoveMonster(shieldMonster);			
+		}
+
+
+
+		/// <summary>
+		/// Checks to see if the Shield collides with any enemies
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void checkShieldCollision(Skill skill, ICombatEntity caster, Mob shieldMonster, CancellationTokenSource cancelToken)
+		{
+			// We have to track sr manually since this skill can hit targets sequentially
+			List<ICombatEntity> hitTargets = new List<ICombatEntity>();
+			var sr = (int)skill.Properties.GetFloat(PropertyName.SkillSR);
+			if (ZoneServer.Instance.Conf.World.DisableSDR)
+				sr = int.MaxValue;
+
+			while (!cancelToken.IsCancellationRequested)
+			{
+				var splashArea = new Circle(shieldMonster.Position, 40);
+
+				var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+				// you can't hit the same target twice (until the shield is on the way back)
+				if (targets.Count > 0)
+					targets.RemoveAll(hitTargets.Contains);
+
+				var targetsLeft = sr - hitTargets.Count;
+
+				if (targets.Count > targetsLeft)
+					targets.RemoveRange(targetsLeft, targets.Count - targetsLeft);
+
+				if (targets.Count > 0)
+				{
+					hitTargets.AddRange(targets);
+					Attack(skill, caster, targets);
+				}
+
+				await Task.Delay(TimeSpan.FromMilliseconds(50));
+			}			
+		}
+
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="targets"></param>
+		private void Attack(Skill skill, ICombatEntity caster, List<ICombatEntity> targets)
+		{
+			var damageDelay = TimeSpan.FromMilliseconds(100);
+			var skillHitDelay = TimeSpan.Zero;
+
+			var bonusPatk = 0f;
+
+			if (caster.IsBuffActive(BuffId.HighGuard_Abil_Buff) && caster is Character character)
```

Comment:
Again, `TryGetBuff` may simplify this code.

Reply:
It's been replaced with TryGetBuff, this was pretty old code.

Reply:
Has it? I'm still seeing `IsBuffActive` and a component get on my side =o

Reply:
It looks like the one I fixed was a separate instance where TryGetBuff was not used, I've also now fixed this one.